sql/parser/create.go

```diff
@@ -182,37 +187,56 @@ type UniqueConstraint struct{}
 
 // IndexTableDef represents an index definition within a CREATE TABLE
 // statement.
-//
-// TODO(pmattis): This should be broken up into IndexTableDef,
-// UniqueConstraintTableDef and
-// PrimaryKeyConstraintTableDef. {Unique,PrimaryKey}ConstraintTableDef would
-// embed an IndexTableDef. Eventually we might have a CheckConstraintTableDef
-// and ForeignKeyConstraintTableDef.
 type IndexTableDef struct {
-	Name       Name
-	PrimaryKey bool
-	Unique     bool
-	Columns    NameList
-	Storing    NameList
+	Name    Name
+	Columns NameList
+	Storing NameList
+}
+
+func (node *IndexTableDef) setName(name Name) {
+	node.Name = name
 }
 
 func (node *IndexTableDef) String() string {
 	var buf bytes.Buffer
+	_, _ = buf.WriteString("INDEX ")
+	if node.Name != "" {
+		fmt.Fprintf(&buf, "%s ", node.Name)
+	}
+	fmt.Fprintf(&buf, "(%s)", node.Columns)
+	if node.Storing != nil {
+		fmt.Fprintf(&buf, " STORING (%s)", node.Storing)
+	}
+	return buf.String()
+}
+
+// ConstraintTableDef represents a constraint definition within a CREATE TABLE
+// statement.
+type ConstraintTableDef interface {
+	TableDef
+	// Placeholder function to ensure that only desired types
+	// (*ConstraintTableDef) conform to the ConstraintTableDef interface.
+	constraintTableDef()
+}
+
+func (*UniqueConstraintTableDef) constraintTableDef() {}
+
+// UniqueConstraintTableDef represents a unique constraint within a CREATE
+// TABLE statement.
+type UniqueConstraintTableDef struct {
+	IndexTableDef
+	PrimaryKey bool
```

Comment:
FYI, I originally created a separate `PrimaryKeyConstraintTableDef` struct, but there was huge overlap with that struct and `UniqueConstraintTableDef`.